pkg/controllers/machine_controller.go

```diff
@@ -329,6 +330,11 @@ func (r *MachineReconciler) processMachineDeletion(ctx context.Context, log logr
 		return fmt.Errorf("failed to delete machine: %w", err)
 	}
 	log.V(1).Info("Deleted machine")
+	machine.Status.State = api.MachineStateTerminated
```

Comment:
What about moving `machine.Status.State = api.MachineStateTerminated` down to the removal of the finalizers?

Reply:
if you think line 353, i don't think it is correct. Terminated state mean machine was deleted and cannot be start again. If you will move it down, machine will be in state terminating and it can be stuck on error during removing of leftovers (volumes, nics, ...) and you will report to customer state terminating which mean machine is in shutdown process what isn't true.